pkg/buildpack/buildpack.go

```diff
@@ -48,22 +52,48 @@ type Blob interface {
 	Open() (io.ReadCloser, error)
 }
 
+type BlobOption func(*blobOption) error
```

Comment:
Adding an option when the Blob is created was the best way I found to modify the behavior without affecting a bunch of places in the code. Only where is required we can create flattened blobs